app/data/ContributionData.scala

```diff
@@ -14,7 +14,31 @@ import play.api.db.Database
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.Try
 
-class ContributionData(db: Database)(implicit ec: ExecutionContext) extends TagAwareLogger {
+trait ContributionData {
+  def insertPaymentHook(paymentHook: PaymentHook)(implicit tags: LoggingTags): EitherT[Future, String, PaymentHook]
+  def insertPaymentMetaData(pmd: ContributionMetaData)(implicit tags: LoggingTags): EitherT[Future, String, ContributionMetaData]
+  def saveContributor(contributor: Contributor)(implicit tags: LoggingTags): EitherT[Future, String, Contributor]
+}
+
+class CurrentAndNewContributionData(currentDB: ContributionDataInstance, newDB: ContributionDataInstance) extends ContributionData {
+
+  override def insertPaymentHook(paymentHook: PaymentHook)(implicit tags: LoggingTags): EitherT[Future, String, PaymentHook] = {
+    currentDB.insertPaymentHook(paymentHook)
+    newDB.insertPaymentHook(paymentHook)
+  }
+
+  override def insertPaymentMetaData(pmd: ContributionMetaData)(implicit tags: LoggingTags): EitherT[Future, String, ContributionMetaData] = {
+    currentDB.insertPaymentMetaData(pmd)
+    newDB.insertPaymentMetaData(pmd)
+  }
+
+  override def saveContributor(contributor: Contributor)(implicit tags: LoggingTags): EitherT[Future, String, Contributor] = {
+    currentDB.saveContributor(contributor)
+    newDB.saveContributor(contributor)
+  }
+}
+
+class ContributionDataInstance(db: Database)(implicit ec: ExecutionContext) extends ContributionData with TagAwareLogger {
```

Comment:
if the database name could be accessed (e.g. by an additional parameter in the class constructor, or a method on the `Database` class (?)), then we could reference this in the log messages. This may prove useful for debugging purposes - especially as the calls are asynchronous which means the data base can not be inferred from the order of the log messages.

Reply:
database has a `name` property which, in this case, will be [TEST/LIVE] or [TEST/LIVE]_auroroa. Good shout on the logging

Reply:
👌